Adjust documentation for destination look via issuerlinter prettier (#89)
SAP/cloud-sdk: docs/js/features/connectivity/destination.mdx

- `OAuth2ClientCredentials`
- `ClientCertificateAuthentication`

> The SDK automatically parses the response of the destination service and uses the provided authentication information for the request to the target system.

Reviewer: ```suggestion
The SAP Cloud SDK automatically parses the response of the destination service and uses the provided authentication information for the request to the target system.
```

Reviewer: This should make Vale happy

These destinations are called `instance destinations` since they are tied to a service binding called instance on SCP.
In every request, these destinations are added to the destinations returned by the destination service.
:::

#### Destination Lookup without a JWT

> There are situations where you do not have a JWT issued by the XSUAA but need to lookup a destination e.g. background processes.

Reviewer 2: I think lookup is not a verb, but look up is.
```suggestion
There are situations where you do not have a JWT issued by the XSUAA but need to look a destination up e.g. in background processes.
```

In such situations the property `iss` of the `DestinationAccessorOptions` can be used:

```js

> .execute({destinationName: 'myDestination'},{iss:yourIssuerValue})

Reviewer 2: ```suggestion
.execute(
  { destinationName: 'myDestination' },
  { iss:yourIssuerValue }
 )
```

```

The value for `iss` is supposed to be the same as in a JWT issued from the XSUAA e.g. `https://yourSubdomain.localhost:8080/uaa/oauth/token`.

> In principle only the host of the url is relevant but since the same parsing and replace methods are used for the `iss` and `userJWT` case please provide the URL as stated before.

Reviewer 2: ```suggestion
In principle only the host of the url is relevant but since the same parsing and replacement methods are used as for the JWT handling, the URL has to be provided in the format above.
```

These destinations are called `instance destinations` since they are tied to a service binding called instance on SCP.
In every request, these destinations are added to the destinations returned by the destination service.
:::

#### Destination Lookup without a JWT

There are situations where you do not have a JWT issued by the XSUAA but need to look a destination up e.g. in background processes.
In such situations the property `iss` of the `DestinationAccessorOptions` can be used:

> ```js
> .execute(
>   { destinationName: 'myDestination' },
>   { iss:yourIssuerValue }
>  )
> ```

Reviewer: ```suggestion
```js
.execute(
  { destinationName: 'myDestination' },
  { iss: yourIssuerValue }
 )
```
```

> The value for `iss` is supposed to be the same as in a JWT issued from the XSUAA e.g. `https://yourSubdomain.localhost:8080/uaa/oauth/token`.
> In principle only the host of the url is relevant but since the same parsing and replacement methods are used as for the JWT handling, the URL has to be provided in the format above.

Reviewer: ```suggestion
The value for `iss` is supposed to be the same as in a JWT issued from the XSUAA e.g. `https://yourSubdomain.localhost:8080/uaa/oauth/token`.
In principle only the host of the URL is relevant but since the same parsing and replacement methods are used for the JWT handling, the URL has to be provided in the format above.
```

> :::note
> If a JWT is used in the destination lookup a validation of the provided token is performed.
> This validation ensures that the JWT has not been manipulated.
> If only the `iss` is provided no such validation is performed.
> Note that the given subdomain value defines from which tenant destinations are fetched.
> So a wrong value may break the tenant isolation.
> It is your responsibility as a developer to ensure that the provided value for the `iss` property is valid.
> :::

Reviewer: ```suggestion
:::note
If a JWT is used in the destination lookup a validation of the provided token is performed.
This validation ensures that the JWT has not been manipulated.
If only the `iss` is provided no such validation is performed.
Note that the given subdomain value defines from which tenant destinations are fetched.
So a wrong value may break the isolation for tenants.
It is your responsibility as a developer to ensure that the provided value for the `iss` property is valid.
:::
```
